### PR TITLE
Make more values configurable

### PR DESCRIPTION
**Pull request by LordTermor**

### CODE REVIEW

File: conanfile.txt [hunk 1]
```diff
@@ -31,4 +31,4 @@ boost:shared=True
 c-ares:shared=True
 
 [imports]
-lib, *.so* -> ./bin
+lib, *.so* -> ./bin/libs/
```
Comment (romangg) [inline conanfile.txt:34]: What is `/bin/libs`? There is either `/usr/bin` or `/usr/libs`. Or are we talking about the build directory here?

File: docker/prod.Dockerfile [hunk 2]
```diff
@@ -5,7 +5,7 @@ COPY ./ /src
 RUN cmake -B/build -S/src -DCMAKE_BUILD_TYPE=Release -G Ninja
 RUN cmake --build /build
 RUN cp /src/daemon/box.yml /build/bin/
-ENV LD_LIBRARY_PATH=./
+ENV LD_LIBRARY_PATH=./libs/
```
Comment (romangg) [inline docker/prod.Dockerfile:8]: Setting the `LD_LIBRARY_PATH` usually isn't necessary when we install something with CMake. There is RPATH for that too.

File: daemon/application.cpp [hunk 3]
```diff
@@ -46,6 +52,18 @@ void setup_logger() {
...
 }
 
+toml::table setup_toml_configuration(const std::filesystem::path& config_path) {
+    if (!std::filesystem::exists(config_path)) {
+        std::ofstream stream {config_path};
+    }
+
+    auto result = toml::parse_file(config_path.string());
+
+    if (!result) { return toml::table {}; }
```
Comment (romangg) [inline daemon/application.cpp:62]: That's not nice to read.

File: daemon/application.cpp [hunk 4]
```diff
@@ -29,9 +30,14 @@
 #include <drogon/HttpAppFramework.h>
 #include <filesystem>
 #include <kangaru/debug.hpp>
+#include <lmdbxx/lmdb++.h>
 #include <memory>
+#include <system_error>
 #include <vector>
 
+#define TOML_EXCEPTIONS 0
```
Comment (romangg) [inline daemon/application.cpp]: You use exceptions in for LMDB. Why not for toml lib?

File: daemon/application.cpp [hunk 5]
```diff
@@ -71,25 +101,39 @@ void setup_di_container(kgr::container& ctr) {
...
 
     ctr.service<di::Persistence::SectionRepository>();
     ctr.emplace<di::Persistence::UserRepository>(std::string("bxt::Users"));
     ctr.emplace<di::Persistence::PackageLogEntryRepository>(
         std::string("bxt::PackageLogs"));
 
-    ctr.emplace<di::Persistence::Box::Pool>("./box/pool/");
-    ctr.emplace<di::Persistence::Box::LMDBPackageStore>("./box/", "bxt::Box");
+    ctr.emplace<di::Persistence::Box::Pool>();
+    ctr.emplace<di::Persistence::Box::LMDBPackageStore>("bxt::Box");
```
Comment (romangg) [inline daemon/application.cpp:139]: What does this stuff do? Can you add a comment?

File: daemon/utilities/lmdb/LMDBOptions.h [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+#pragma once
+
+#include "utilities/configuration/Configuration.h"
+
+#include <filesystem>
+namespace bxt::Utilities::LMDB {
+
+struct LMDBOptions : public bxt::Utilities::Configurable {
```
Comment (romangg) [inline daemon/utilities/lmdb/LMDBOptions.h:15]: Add a comment what this struct does.

**Review by romangg — COMMENTED** [review]
Please add some comments to give me a better high-level understanding of the changes.

The code changes feel quite complex. I believe most of it is due to the DI-framework, is it?

File: conanfile.txt (hunk 1 above)
Comment (LordTermor) [reply conanfile.txt:34]: Yep, all libs from conan are going into the bin folder with a built binary.

File: daemon/application.cpp (hunk 4 above)
Comment (LordTermor) [reply daemon/application.cpp]: I tend to use exception-less design using `expected`. Exceptions are there only for libs that don't support any other way (e.g. lmdb++, yes)

File: daemon/utilities/lmdb/LMDBOptions.h (hunk 6 above)
Comment (LordTermor) [reply daemon/utilities/lmdb/LMDBOptions.h:15]: That's pretty much a part of conan's howto. We can use rpath I think but I'm still aiming only for docker instances so there's very small reason to do so.

https://docs.conan.io/en/1.60/howtos/manage_shared_libraries/rpaths.html

File: daemon/application.cpp (hunk 5 above)
Comment (LordTermor) [reply daemon/application.cpp:139]: Which part exactly? That's a DI container setup.

File: docker/prod.Dockerfile (hunk 2 above)
Comment (LordTermor) [reply docker/prod.Dockerfile:8]: That's pretty much a part of conan's howto. We can use rpath I think but I'm still aiming only for docker instances so there's very small reason to do so.

https://docs.conan.io/en/1.60/howtos/manage_shared_libraries/rpaths.html

**Review by romangg — APPROVED** [review]

**Pull request merged**